crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -16,6 +16,15 @@ use crate::context_data::package_cache::PackageCache;
 use crate::error::Error;
 use sui_types::object::Object as NativeObject;
 
+/// Dynamic fields are heterogeneous fields that can be added or removed at runtime,
+/// and can have arbitrary user-assigned names.
+/// There are two sub-types of dynamic fields:
+/// 1) Dynamic Fields can store any value that has the store ability,
+/// however an object stored in this kind of field will be considered wrapped and will not be
+/// accessible directly via its ID by external tools (explorers, wallets, etc) accessing storage.
+/// 2) Dynamic Object Fields values must be Sui objects (have the key and store abilities,
+/// and id: UID as the first field), but will still be directly accessible via
+/// their object ID after being attached.
```

Comment:
Some consistent whitespace and markdown formatting would help readability here (and in generated docs).

```suggestion
/// Dynamic fields are heterogeneous fields that can be added or removed at runtime,
/// and can have arbitrary user-assigned names. There are two sub-types of dynamic 
/// fields:
///
/// 1) Dynamic Fields can store any value that has the `store` ability, however an object 
///    stored in this kind of field will be considered wrapped and will not be accessible 
///    directly via its ID by external tools (explorers, wallets, etc) accessing storage.
/// 2) Dynamic Object Fields values must be Sui objects (have the `key` and `store` 
///    abilities, and id: UID as the first field), but will still be directly accessible off-chain 
///    via their object ID after being attached.
```

Also, note that Dynamic Object Fields are only accessible off-chain by their ID -- it's not possible to construct a transaction that accepts a dynamic object field value by ID as input, after it's attached.

Reply:
Thanks for the extra note.